README.md

```diff
@@ -1,555 +1,3169 @@
+unsuccessful.
+
+```dart
+Future<Todo> fetchTodos() async {
+  final response = await http
+      .get(Uri.parse('https://jsonplaceholder.typicode.com/todos/1'));
+
+  if (response.statusCode == 200) {
+    // If the server did return a 200 OK response,
+    // then parse the JSON.
+    return Todo.fromJson(jsonDecode(response.body));
+  } else {
+    // If the server did not return a 200 OK response,
+    // then throw an exception.
+    throw Exception('Failed to load todos');
+  }
+}
+```
 
-Follow the steps to add the Firebase SDK / Go to people API:<br />
+How would we call this inside a widget? 
+We could do this inside `initState()`! 
+It is called exactly one time and never again!
+Do **not** put an API call in the `build()` method
+(unless you know what you are doing). 
+This method is called every time a render occurs,
+which is quite often!
+
+```dart
+class _MyAppState extends State<MyApp> {
+  late Future<Todo> futureTodo;
+
+  @override
+  void initState() {
+    super.initState();
+    futureTodo = fetchTodo();
+  }
+  // ···
+}
+```
 
-https://developers.google.com/people/v1/getting-started
+Finally, to display the data, we would want to use the
+`FutureBuilder` widget. As the name implies, it's a 
+widget made to handle async data operations. 
+
+```dart
+FutureBuilder<Todo>(
+  future: futureTodo,
+  builder: (context, snapshot) {
+    if (snapshot.hasData) {
+      return Text(snapshot.data!.title);
+    } else if (snapshot.hasError) {
+      return Text('${snapshot.error}');
+    }
+
+    // By default, show a loading spinner.
+    return const CircularProgressIndicator();
+  },
+)
+```
+
+The `future` paramter relates to object we want to work with.
+In this case, it is a parsed `Todo` object. 
 
-Follow those 3 steps.
+The `builder` function tells Flutter what needs to be rendered, 
+depending on the current state of `Future`, which can 
+be *loading*, *success* or *error*. 
+Depending on the result of the operation, we
+either show the error, the data or a loading animation
+while we wait for the http request to fulfill.
 
-![Flutter](https://i.imgur.com/ro6RiCJ.png)
+Isn't it easy? =)
 
-After that go to credentials and click the user data option.<br />
-Click on the API and go for Android or IOS.<br />
-Name your app again, put the SHA-1 you got from 'gradlew' and the package name you can find in your 'AndroidManifest.xml'.<br />
+## Local databases
+Sometimes, when writing an app, we need to persist
+and query large amounts of data on the local device. 
+In these cases, it is beneficial considering 
+using a database instead of a local file or a key-value store. 
 
+In this walkthrough, we are going to present
+two alternatives: SQLite and ObjectBox.
 
-![Flutter](https://i.imgur.com/wkLI4L0.png)
+### SQLite 
 
+SQLite is one of the most popular methods for storing data locally.
+For this demo, we will use the package 
+[`sqflite`](https://pub.dev/packages/sqflite).
```

Comment:
Not a fan of using **`sqflite`** in our tutorial because, as noted in: https://github.com/dwyl/learn-flutter/issues/70 we will be using **`drift`** because it has _muuuch_ better features, Web Support and extensive testing.

**`sqflite`** not so much ... https://pub.dev/packages/sqflite

<img width="1062" alt="sqflite-no-web-build-failing" src="https://user-images.githubusercontent.com/194400/203290360-08a3ea9c-975d-44a0-a6ee-3d7ec206f0c6.png">

Reply:
I tried to keep the original structure, hence why I did a rundown with `sqlite`. 
`Drift` actually uses `sqlite` under the hood.

Do you want me to add a section for `Drift` or...?

Reply:
Indeed. `drift` uses `drift_sqflite` which in turn depends on `sqflite` [drift_sqflite/pubspec.yaml#L13](https://github.com/simolus3/drift/blob/df7645781985daa7ba44b154f158d79e34e719aa/drift_sqflite/pubspec.yaml#L13) 👀 
but appears to add quite a few enhancements ... ✨ 


We don't need to update the tutorial _yet_. ⏳ 
I just want to finish reading through this. 🙏 
But left a comment as a reminder. 💭